docs/src/main/asciidoc/extension-maturity-matrix.adoc

```diff
@@ -0,0 +1,178 @@
+
+=== Virtual thread support
+
+Not every library is suitable for using with virtual threads, out of the box.
+xref:virtual-threads#why-not["Why not virtual threads everywhere?"] explains why.
+
+To get your library working properly with virtual threads, you should make sure the library is not pinning the carrier thread.
+ Quarkus has xref:virtual-threads.adoc#testing-virtual-thread-applications[test helpers to do these checks in an automated way].
+ For dispatching work, you should use the xref:virtual-threads.adoc#inject-the-virtual-thread-executor[virtual executor managed by Quarkus]. The link:https://quarkus.io/extensions/io.quarkus/quarkus-websockets-next/[WebSockets-next extension] uses the virtual dispatcher and smart dispatch, and is a good example to follow.
+
+=== Hot path performance optimization
+
+Although Quarkus offers some unique opportunities for extension performance, extension developers shouldn't forget https://www.linkedin.com/pulse/how-optimize-software-performance-efficiency-subcodevs/[the basics of performance optimization].
+
+=== Non-blocking internals
+
+Quarkus's reactive core is a key contributor to its excellent throughput and scalability. Extensions should consider adopting this model for their own internal operations.
+
+=== Add Mutiny-based APIs
+
+For maximum scalability, go beyond the reactive core and enable fully reactive programming, using Mutiny. Most projects that support a reactive programming model offer two distinct extensions, a `-reactive` and a plain one.
+See, for example, https://quarkus.io/extensions/io.quarkiverse.quarkus-elasticsearch/quarkus-elasticsearch/[ElasticSearch] and https://quarkus.io/extensions/io.quarkiverse.quarkus-elasticsearch-reactive/quarkus-elasticsearch-reactive/[ElasticSearch Reactive] extensions.
+
+== Operations
+
+Developer joy is important, but so are observability, maintainability, and other operational considerations.
+Many of these characteristics come by default with the Quarkus framework or https://quarkus.io/extensions/io.quarkus/quarkus-opentelemetry/[observability-focussed extensions]. But extensions can do more.
+
+=== Logging
+
+Quarkus uses JBoss Logging as its logging engine, and xref:logging[supports several logging APIs]. (This is normal Java logging, not OpenTelemetry logging.)
+
+Avoid using errors and warnings for conditions that will not affect normal operation. These outputs can cause false alarms in user monitoring systems.
+
+
+=== Define health endpoints
+
+Extensions may wish to xref:writing-extensions#extension-defined-endpoints[define library-specific endpoints] for health criteria which are specific to that extension. To add a new endpoint, extensions should produce a `NonApplicationRootPathBuildItem`.
+
+=== Tracing context
+
+You should test that OpenTelemetry output for applications using your extension have properly-defined spans. You may need to do extra work to ensure spans are created with the right OpenTracing ID.
```

Comment:
Should be tracing ID, not OpenTracing ID

Reply:
Done, thank you!